internal/controllers/gateway/gateway_utils.go

```diff
@@ -670,3 +666,71 @@ func isTLSSecretValid(secret *corev1.Secret) bool {
 	}
 	return true
 }
+
+func routeAcceptedByGateways(routeNamespace string, parentStatuses []RouteParentStatus) []k8stypes.NamespacedName {
```

Comment:
Since the return type here doesn't say what type it is for, I'd definitely add at the very least a comment, stating what this function is all about and what it is it's returning.